src/main/kotlin/no/nb/mlt/wls/order/service/OrderService.kt

```diff
@@ -65,6 +67,53 @@ class OrderService(val db: OrderRepository, val synqService: SynqOrderService) {
         return ResponseEntity.status(HttpStatus.CREATED).body(order.toApiOrderPayload())
     }
 
+    suspend fun updateOrder(payload: ApiOrderPayload): ResponseEntity<ApiOrderPayload> {
+        throwIfInvalidPayload(payload)
+
+        val existingOrder =
+            db.findByHostNameAndHostOrderId(payload.hostName, payload.orderId)
+                .timeout(Duration.ofSeconds(8))
+                .onErrorMap {
+                    if (it is TimeoutException) {
+                        logger.error(it) {
+                            "Timed out while fetching from WLS database. Relevant payload: $payload"
+                        }
+                    } else {
+                        logger.error(it) { "Unexpected error for $payload" }
+                    }
+                    ServerErrorException("Failed while checking if order already exists in the database", it)
+                }
+                .awaitSingleOrNull()
```

Comment:
I think it's better to extract that into a method. I have done it in my branch, just need to remember to replace these calls when we merge them.

Reply:
Same